portfolio/src/main/java/com/google/sps/servlets/UsersServlet.java

```diff
@@ -0,0 +1,50 @@
+
+package com.example.appengine.users;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+// With @WebServlet annotation the webapp/WEB-INF/web.xml is no longer required.
+@WebServlet(
+    name = "UserAPI",
+    description = "UserAPI: Login / Logout with UserService",
+    urlPatterns = "/userapi"
+)
+public class UsersServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    String thisUrl = req.getRequestURI();
+
+    resp.setContentType("text/html");
+    if (req.getUserPrincipal() != null) {
+      resp.getWriter()
+          .println(
+              "<p>Hello, "
+                  + req.getUserPrincipal().getName()
```

Comment:
is |getName| guaranteed to always return a value? What if the name isn't set?

Reply:
The getName is the email address you sign in with.

Reply:
Yeah, the name is always present as long as the user principal is. Doesn't really make sense to have a user without a username.